backend/uclapi/workspaces/views.py

```diff
@@ -351,3 +354,35 @@ def get_live_map(request, *args, **kwargs):
     response["Content-Length"] = len(map_svg)
 
     return response
+
+
+class SurveysList(generics.ListAPIView):
+    queryset = Surveys.objects.all()
+    serializer_class = SurveysSerializer
+    filterset_fields = {"survey_id": ["exact"], "active": ["exact"]}
+
+    @method_decorator(uclapi_protected_endpoint(personal_data=False, last_modified_redis_key="Workspaces-Historical"))
+    def list(self, request, *args, **kwargs):
+        return pretty_response(super().list(request, *args, **kwargs), custom_header_data=kwargs)
+
+
+class SensorsList(generics.ListAPIView):
+    queryset = Sensors.objects.all()
+    serializer_class = SensorsSerializer
+    filterset_fields = {"survey_id": ["exact"], "sensor_id": ["exact"]}
+
+    @method_decorator(uclapi_protected_endpoint(personal_data=False, last_modified_redis_key="Workspaces-Historical"))
+    def list(self, request, *args, **kwargs):
+        return pretty_response(super().list(request, *args, **kwargs), custom_header_data=kwargs)
+
+
+class HistoricalList(generics.ListAPIView):
+    queryset = Historical.objects.all()
+    serializer_class = HistoricalSerializer
+    filterset_fields = {"survey_id": ["exact"], "sensor_id": ["exact"],
+                        "datetime": ["gte", "lte", "exact", "gt", "lt"]}
```

Comment:
This may simply be answered by just looking at the docs but that's too painful :p

1. Does this mean that it accepts a range of `datetime` or it simply (0 -> datetime) (datetime -> now)? If so, I believe a range if preferable (or at least should be an option?).
2. Should we make `survey_id` and `sensor_id` compulsory filters? This should help with `CursorPagination` (which has better asymptotic properties) and also will reduce the data we send by half aproximately (we don't have to send `sensor_id`/`survey_id` for each delta, as we know they'll be the same).

Reply:
1. This allows a range of datetimes, for example ```{"datetime__gte": "2020-01-01 00:00:00", "datetime__lte", "2020-01-02 00:00:00"}``` for the data between 2020-01-01 and 2020-01-02 (inclusive).
2. I have made survey_id a compulsory field it significantly speeds up pagination (in most scenarios I imagine the developer querying a single location and trying to find all the seats in a 24 hour period which is faster if they don't need to provide the sensor_id too).

Reply:
1. Great
2. What's the speedup if `sensor_id` is compulsory too (also remove `sensor/survey_id` from the response in that benchmark)? Otherwise, the reasoning here is sound, so I'll keep it if the performance properties of making `sensor_id` compulsory aren't too compelling.

Reply:
2. Discussed in meeting, apparently speedup with `sensor_id` as compulsory is minimal so leaving as is